### DISCUSSION TITLE
iPrima shows just 3 first episodes of a TV series, web shows "Došlo k neočekávané chybě, zkuste prosím obnovit stránku"

### DISCUSSION THREAD

**Discussion by slrslr** [sunecz/Media-Downloader#2] [Q&A] [answered]
Other iPrima TV series displays episodes properly in Media Downloader, but this one i am checking does not.
I can see just first 3 episodes. On the web when i check 41. episode of the problematic TV series (this episode is displayed to air like like one month ago), it plays no video and instead displays "Došlo k neočekávané chybě, zkuste prosím obnovit stránku" - in two various web browsers.
Other iPrima TV series plays OK on web and shows episodes properly in Media Downloader.
I had free iPrima plan and now light plan. https://play.iprima.cz/ucet/subskripce/plan

When I turned off AdBlock in the web browser and confirmed the **HTML canvas request followed by DRM, the video loaded without problem in the browser**. But there are only the first 3 episodes in the Media Downloader and DRM should hopefully be installed on my Linux:
`mkdir -p resources/binary/drm/ resources/plugin;wget -Omp4decrypt.zip "https://www.bok.net/Bento4/binaries/Bento4-SDK-1-6-0-641.x86_64-unknown-linux.zip";unzip -p mp4decrypt.zip Bento4-SDK-1-6-0-641.x86_64-unknown-linux/bin/mp4decrypt > resources/binary/drm/mp4decrypt;chmod +x resources/binary/drm/mp4decrypt;wget "https://app.sune.tech/mediadown/dwn/etc/plugin.zip";unzip -d resources/plugin plugin.zip;wget "https://app.sune.tech/mediadown/dwn/etc/wv.zip";unzip -d resources/binary/drm wv.zip;chmod +x resources/binary/drm/wv;rm -f mp4decrypt.zip plugin.zip wv.zip`
(started in the Media Downloader folder and was successful, and since then Voyo has been downloading videos with no problem even if videos require decoding. But with iPrima there is the above mentioned problem..)

Any idea what to do or when to do it so i do not have this issue?

**Comment by sunecz** [comment]
Hello,

I checked it out in my browser (Firefox 120.0.1, Windows 10) and I can see all the episodes (1-42). I also checked it in the application (00.02.09-dev.13) and I've got the same results, i.e. all 42 episodes are available.

I have the Premium tier, not the Light one, so I don't know how it behaves with that one, but I can confirm that the Free tier does not work since it is not available for that tier.

I also checked it on Ubuntu (20.04.1). In Firefox 109 a popup was displayed saying that the browser is not supported and the site was a little bit broken, so I updated the Firefox browser (for apt: `sudo apt install --only-upgrade firefox`) and in the version 120.0.1 (same as on Windows), it is showing all 42 episodes. After I enabled DRM and Firefox auto-installed some components, the video is also playable. So if it is not working for you in your browser, do you have the latest version?

I also checked the application without login credentials and it only shows the first 3 episodes.

So in conclusion:
- If you do NOT have credentials specified in the application's settings, there will be only the first 3 episodes, since they are the only ones available for the Free tier.
- If you DO have credentials specified in the application's settings, they are valid and the Premium tier is active for them, it displays all 42 episodes.

So I guess that the TV show is not available for the Light tier? But it does not make much sense, since the only aspect that should be different to the Premium tier is the video quality (HD instead of FullHD).

Do you have login credentials specified in the application's settings and are they valid?

**Reply by slrslr** [reply]
Thank you @sunecz In case of my iPrima Light subscription, i can play the linked 41. episode in certain web browser, but the problem is inside Media Downloader. When i check that i have iprima login credentials inside Media Downloader, the credentials window shows "Přihlašování" for a few seconds, then disappear and i am unable to click, because there is some overlay error window with unknown content since it is like one pixel wide as seen on the image a window switcher:
![media downloader hidden error](https://github.com/sunecz/Media-Downloader/assets/6596726/ae0b53c6-c060-4cba-b71e-c3a1838fb5f7)
This is Linux with openjdk 11.0.21. So i hit Esc to close unknown error window (if it can tell you something useful, it would be handy if there is way to somehow log the error). So iprima credentials are filled. But doubleclicking "Souboj s láskou" TV series to show episodes, cause exception window:

```
	java.lang.NullPointerException
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Devices.create(PrimaAuthenticator.java:371)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Devices.createDefault(PrimaAuthenticator.java:375)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Cached.getDevice(PrimaAuthenticator.java:259)
	at sune.app.mediadown.concurrent.VarLoader.valueRaw(VarLoader.java:62)
	at sune.app.mediadown.concurrent.VarLoader.valueChecked(VarLoader.java:73)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Cached.device(PrimaAuthenticator.java:274)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator.initSessionData(PrimaAuthenticator.java:183)
	at sune.app.mediadown.concurrent.VarLoader.valueRaw(VarLoader.java:62)
	at sune.app.mediadown.concurrent.VarLoader.valueChecked(VarLoader.java:73)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator.sessionData(PrimaAuthenticator.java:193)
	at sune.app.mediadown.media_engine.iprima.PrimaPlus$API.logIn(PrimaPlus.java:253)
	at sune.app.mediadown.media_engine.iprima.PrimaPlus$API.initSessionHeaders(PrimaPlus.java:242)
	at sune.app.mediadown.concurrent.VarLoader.valueRaw(VarLoader.java:62)
	at sune.app.mediadown.concurrent.VarLoader.valueChecked(VarLoader.java:73)
	at sune.app.mediadown.media_engine.iprima.PrimaPlus$API.logInHeaders(PrimaPlus.java:264)
	at sune.app.mediadown.media_engine.iprima.PrimaPlus$API.lambda$getEpisodes$3(PrimaPlus.java:362)
	at sune.app.mediadown.task.ListTask$1.run(ListTask.java:34)
	at sune.app.mediadown.task.Task.doRun(Task.java:42)
	at java.base/java.lang.Thread.run(Unknown Source)

```

after closing that exception, Media Downloader showing only 3 episodes despite i have Light subscription on the web
![image](https://github.com/sunecz/Media-Downloader/assets/6596726/0b384d08-0ea2-4cd7-b0bc-893c8502850f)
and being able to play episodes on the web without problem. So i guess Media Downloader is unable to log me in and so displays only 3 Free episodes.

**Reply by sunecz** [reply]
The fact that there are no profiles being shown in the Credentials window is weird and should not happen if the credentials present are correct. For whatever reason the actual content in password field may be "corrupted" (or however you want to call it). I've encountered that myself once and was unable to replicate it ever since.

Please, try the following:
- Go to Credentials > iPrima and clear both the email and the password field, then save the credentials.
- Restart the application.
- Go to Credentials > iPrima and wait for it to log in. The application should use the default credentials and after a while there should be the following profiles available in the select field: Automatically and Media. If that's the case, all is OK.
- Optional: You can try to get media sources of the first episode of the TV show to just ensure that only the Free tier qualities are available, since the default account is only Free tier.
- In the Credentails window, enter your credentials to the empty fields and save them.
- Restart the application.
- Go to Credentials > iPrima and wait for it to log in. It should work now (hopefully).

Just a sidenote: The issue with incorrectly displaying the error window is some bizare graphical issue on Linux-based systems that I have no idea why is caused. It may be some issue in the JavaFX framework, since the application uses only the latest publicly available version 11.0.2.

**Reply by slrslr** [reply]
I have followed yours suggested steps except optional one, but that not worked. I continue seeing only first 3 episodes despite having Light subscription and on the iprima login credentials editing window, it continue to show either that one pixel bizzare window or the exception

```
java.lang.NullPointerException
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Devices.create(PrimaAuthenticator.java:371)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Devices.createDefault(PrimaAuthenticator.java:375)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Cached.getDevice(PrimaAuthenticator.java:259)
	at sune.app.mediadown.concurrent.VarLoader.valueRaw(VarLoader.java:62)
	at sune.app.mediadown.concurrent.VarLoader.valueChecked(VarLoader.java:73)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator$Cached.device(PrimaAuthenticator.java:274)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator.initSessionData(PrimaAuthenticator.java:183)
	at sune.app.mediadown.concurrent.VarLoader.valueRaw(VarLoader.java:62)
	at sune.app.mediadown.concurrent.VarLoader.valueChecked(VarLoader.java:73)
	at sune.app.mediadown.media_engine.iprima.PrimaAuthenticator.sessionData(PrimaAuthenticator.java:193)
	at sune.app.mediadown.media_engine.iprima.IPrimaEnginePlugin$IPrimaCredentialsType$ProfileSelect.lambda$loadItems$5(IPrimaEnginePlugin.java:277)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

```

But clearing the login fields and saving indeed caused what you have described: "there should be the following profiles available in the select field: Automatically and Media."

**Reply by sunecz** [reply]
Thank you for checking it out!

Based on the error it is giving you, there is an issue when creating a new device for the application. I cannot test it directly, since I don't have access to your account, but I can send you a modified version of the plugin that should display more information, if you want.

I don't want to post it here though, so if you want, can you contact me at my email I have on the application's website? I'll prepare the modified version.

**Reply by slrslr** [reply, marked as answer]
🇺🇸:
The issue was caused by iPrima 5 devices limit https://ucet.iprima.cz/zarizeni and Media Downloader not made to report it to user. After removing some devices (see the link above), Media Downloader started showing all episodes. The issue where one is unable to play video inside web browser is caused by unsupported web browser or some extension blocking DRM, HTML canvas or similar browser functions.
🇨🇿
Problém byl způsoben limitem 5 zařízení iPrima https://ucet.iprima.cz/zarizeni a Media Downloaderem, který nebyl naprogramován, aby to uživateli oznámil. Po odstranění některých zařízení (viz. odkaz výše) začal Media Downloader zobrazovat všechny epizody. Problém, kdy člověk nemůže přehrát video uvnitř webového prohlížeče, je způsoben nepodporovaným webovým prohlížečem nebo nějakým rozšířením blokujícím DRM, HTML canvas nebo podobné funkce prohlížeče.
